pkg/controllers/namespace/namespace_controller.go

```diff
@@ -0,0 +1,242 @@
+  - If the ns do not have either of the annotation above, then we believe it is using default VPC, try to search
+    default VPC in network config CR store. The default VPC network config CR's name is "default".
+*/
+func (r *NamespaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	obj := &v1.Namespace{}
+	log.Info("reconciling K8s namespace", "namespace", req.NamespacedName)
+	metrics.CounterInc(r.NSXConfig, metrics.ControllerSyncTotal, common.MetricResTypeNamespace)
+
+	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
+		log.Error(err, "unable to fetch namespace", "req", req.NamespacedName)
+		return common.ResultRequeueAfter5mins, err
+	}
+
+	// processing create/update event
+	ns := obj.GetName()
+	if obj.ObjectMeta.DeletionTimestamp.IsZero() {
+		metrics.CounterInc(r.NSXConfig, metrics.ControllerUpdateTotal, common.MetricResTypeNamespace)
+		log.Info("start processing namespace create/update event", "namespace", ns)
+		ctx := context.Background()
+		annotations := obj.GetAnnotations()
+		r.insertNamespaceNetworkconfigBinding(ns, annotations)
+		// read anno "nsx.vmware.com/vpc_name", if ns contains this annotation, it means it will share
+		// infra VPC, if the ns in the annotation is the same as ns event, create infra VPC, if not,
+		// skip the event.
+		ncName, ncExist := annotations[types.AnnotationVPCNetworkConfig]
+		vpcName, nameExist := annotations[types.AnnotationVPCName]
+		if nameExist {
+			log.Info("read ns annotation vpcName", "VPCNAME", vpcName)
+			res := strings.Split(vpcName, "/")
+			// The format should be namespace/vpc_name
+			if len(res) != 2 {
+				message := fmt.Sprintf("incorrect vpcName annotation %s for namespace %s", vpcName, ns)
+				r.namespaceError(&ctx, obj, message, nil)
+				// If illegal format, skip handling this event?
+				return common.ResultNormal, nil
+			}
+			log.Info("start to handle vpcName anno", "VPCNS", res[1], "NS", ns)
+			if ns == res[0] {
+				// namespace name is the same as ns creation event, create VPC CR
+				// read infra network config cr name which is in annotation nsx.vmware.com/vpc-network-configuration
+				if !ncExist {
+					// if no infra network config cr annotation, skip the event
+					message := fmt.Sprintf("missing network config CR name annotation for namespace %s", ns)
+					r.namespaceError(&ctx, obj, message, nil)
+					return common.ResultNormal, nil
+				}
+				log.Info("creating vpc using customer defined vpc name", "VPCName", res[1])
+				if _, err := r.createVPCCR(&ctx, obj, ns, ncName, &res[1]); err != nil {
+					message := "failed to create vpc CR using given vpc name, requeue namespace create event"
+					r.namespaceError(&ctx, obj, message, err)
+					return common.ResultRequeueAfter10sec, nil
+				}
+				return common.ResultNormal, nil
+			} else {
+				log.Info("name space is using shared vpc, with vpc name anno", "VPCNAME", vpcName, "Namespace", ns)
+			}
```

Comment:
How about change to:
if ns != res[0] {
    log.Info("name space is using shared vpc, with vpc name anno", "VPCNAME", vpcName, "Namespace", ns)
    return common.ResultNormal, nil
}

Then for rest code, use the same logic with following lines 212-220